vm/actor/src/builtin/miner/state.rs

```diff
@@ -1005,6 +1025,100 @@ impl MinerInfo {
     }
 }
 
+#[cfg(feature = "json")]
+pub mod json {
+    use super::*;
+    use address::json::AddressJson;
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "PascalCase")]
+    pub struct MinerBaseInfoJson {
+        #[serde(with = "bigint_ser::json::opt")]
+        pub miner_power: Option<TokenAmount>,
+
+        #[serde(with = "bigint_ser::json::opt")]
+        pub network_power: Option<TokenAmount>,
+
+        pub sectors: Vec<SectorInfoJson>,
+
+        #[serde(with = "address::json")]
+        pub worker_key: Address,
+
+        pub sector_size: SectorSize,
+
+        #[serde(with = "beacon::json")]
+        pub prev_beacon_entry: BeaconEntry,
+
+        pub beacon_entries: Vec<BeaconEntryJson>,
+
+        pub eligible_for_mining: bool,
+    }
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "PascalCase")]
+    pub struct MinerInfoJson {
+        pub owner: AddressJson,
+
+        pub worker: AddressJson,
+
+        #[serde(with = "address::json::opt")]
+        pub new_worker: Option<Address>,
+
+        #[serde(with = "address::json::opt_vec")]
+        pub control_addresses: Option<Vec<Address>>, // Must all be ID addresses.
+
+        pub worker_change_epoch: ChainEpoch,
+
+        /// Optional worker key to update at an epoch
+        pub pending_worker_key: Option<WorkerKeyChange>,
```

Comment:
You shouldn't need to document or expose the fields? Or is there an issue with that?